File: domain/src/main/java/com/ractoc/eve/domain/assets/BlueprintModel.java
package com.ractoc.eve.domain.assets;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import io.swagger.annotations.ApiModel;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@ApiModel(value = "Blueprint Model", description = "Contains the Blueprint model information")
@JsonIgnoreProperties(ignoreUnknown = true)
public class BlueprintModel {
    private int id;
    private String name;
    private double buyPrice;
    private ItemModel item;
    private Long locationId;
    private String locationFlag;
    private int maxProductionLimit;
    private int copyingTime;
    private int inventionTime;
    private int manufacturingTime;
    private int researchMaterialTime;
    private int researchTimeTime;
    private Integer materialEfficiency = 0;
    private Integer timeEfficiency = 0;
    private Integer runs = 1;
    private Integer quantity;
    private double mineralBuyPrice;
    private double mineralSellPrice;
    private double jobInstallationCosts;

    private Set<BlueprintMaterialModel> inventionMaterials = new HashSet<>();
    private Set<BlueprintProductModel> inventionProducts = new HashSet<>();
    private Set<BlueprintSkillModel> inventionSkills = new HashSet<>();

    private Set<BlueprintMaterialModel> manufacturingMaterials = new HashSet<>();
    private Set<BlueprintProductModel> manufacturingProducts = new HashSet<>();
    private Set<BlueprintSkillModel> manufacturingSkills = new HashSet<>();

    public Integer getMaterialEfficiency() {
        if (materialEfficiency == null) {
            return 0;
        } else {
            return materialEfficiency;
        }
    }
}

File: angular/MineBuildSell/src/app/shared/model/blueprint.model.ts
import {ItemModel} from './item.model';

export class BlueprintModel {

  constructor(public id: number,
              public name: string,
              public locationId: number,
              public buyPrice: number,
              public materialEfficiency: number,
              public runs: number,
              public manufacturingMaterials: BlueprintMaterialModel[],
              public mineralBuyPrice: number,
              public mineralSellPrice: number,
              public jobInstallationCosts: number,
              public item: ItemModel) {
  }

}

export class BlueprintMaterialModel {

  constructor(public typeId: number,
              public quantity: number,
              public calculatedTotalQuantity: number,
              public buyPrice: number,
              public sellPrice: number) {
  }

}
